CMakeLists.txt

```diff
@@ -150,7 +150,7 @@ if(UR_FORMAT_CPP_STYLE)
         message(STATUS "Found clang-format: ${CLANG_FORMAT} (version: ${CLANG_FORMAT_VERSION})")
 
         set(CLANG_FORMAT_REQUIRED "15.0")
-        if(NOT (CLANG_FORMAT_VERSION VERSION_EQUAL CLANG_FORMAT_REQUIRED))
+        if(NOT (CLANG_FORMAT_VERSION VERSION_GREATER_EQUAL CLANG_FORMAT_REQUIRED))
```

Comment:
clang-format is not backward compatible in terms of its output, so a newer version can (and often does) output something different than the previous one. Since we want to enforce formatting at CI level and to avoid PRs randomly changing styling depending on the author's clang-version, we require one specific clang-format version. If it's different than what you have in your system, the best approach we found is to use pyenv with the [correct version](https://github.com/oneapi-src/unified-runtime/blob/main/third_party/requirements.txt#L40) installed:
https://oneapi-src.github.io/unified-runtime/core/CONTRIB.html#build-environment